.github/workflows/build.yaml

```diff
@@ -0,0 +1,154 @@
+name: Build Serval
+
+on: [push]
+# on:
+#   push:
+#     tags:
+#       - 'v*'
+
+env:
+      CARGO_TERM_COLOR: always
+
+jobs:
+  release:
+    name: release
+    runs-on: ubuntu-latest
+    outputs:
+      upload_url: ${{ steps.create_release.outputs.upload_url }}
+    steps:
+      - name: create a github release
+        id: create_release
+        uses: actions/create-release@v1
+        env:
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+        with:
+          tag_name: ${{ github.ref }}
+          release_name: serval-mesh ${{ github.ref }}
+          draft: true
+          prerelease: false
+
+  linux:
+    name: build for linux
+    runs-on: ubuntu-latest
+    needs: [release]
+    strategy:
+      matrix:
+        arch: [amd64, arm64]
+        include:
+          - arch: amd64
+            target: x86_64-unknown-linux-gnu
+            cross: false
+          - arch: arm64
+            target: aarch64-unknown-linux-gnu
+            cross: true
+      fail-fast: true
+      max-parallel: 1
```

Comment:
I noticed that when the matrix runs in parallel, neither of the parallel executions can make use of the cargo cache step. This slows down the overall release creation time but lowers the compute minutes needed to get there.
There may be a way to just have a gatekeeper step for the cache step that lets subsequent ones start up as soon as the cache step of the first variant has completed, but that seems like over-engineering for a first implementation. Unless someone has a very easy 1-liner for this, of course :)